### DISCUSSION TITLE
OnMouseOver

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#525] [5. Q&A]
패키지 버전 "version": "1.1.5" 업데이트 했는데요 혹시 

MonoBehaviour Messages 추가(OnAnimatorIK, OnMouseDown, OnMouseDrag, OnMouseUp, OnMouseEnter, OnMouseExit, OnMouseOver, OnMouseUpAsButton)

요거 동작 하나요??

https://studio.zepeto.me/kr/guides/world-115

문서에는 동작 한다고 써있는데 안되는거 같아서요

**Comment by pumfit** [comment]
스크립트를 작성한 후에 원하는 오브젝트에 스크립트를 넣어주면 잘 작동합니다!
오브젝트에 콜라이더가 존재하는지 레이어가 Ignore Raycast로 되어있지는 않은 지 확인해보시면 좋을 것 같아요!
![image](https://user-images.githubusercontent.com/46295539/159823796-d185fcca-f4d1-4d41-8f59-d6e39162e423.png)

**Reply by smilejsu82** [reply]
제페토 템플릿에서 타입스크립트로 작성된 컴포넌트를 사용하셨나요? 

저는 동작하지 않았습니다 

그래서 제페토 템플릿이 아닌 새로운 프로젝트를 만들어 테스트 해봤습니다

이것은 동작했습니다 

제페토 템플릿에서 다시 테스트 해보겠습니다 

답변 감사합니다

**Comment by smilejsu82** [comment]
![Screen Shot 2022-03-24 at 오전 10 50](https://user-images.githubusercontent.com/50188795/159826216-11202f47-4def-4d40-aed9-7ef47012405d.png)
![Screen Shot 2022-03-24 at 오전 10 50 - 2](https://user-images.githubusercontent.com/50188795/159826230-c0b2e1cb-9724-4933-95a2-94be1d150b58.png)

https://user-images.githubusercontent.com/50188795/159826236-122d1c97-2c7b-4357-addf-917a988b797a.mp4

![Screen Shot 2022-03-24 at 오전 10 52](https://user-images.githubusercontent.com/50188795/159826242-07966fb6-e1d8-4427-ab12-96f15edd1772.png)
![Screen Shot 2022-03-24 at 오전 10 52 - 2](https://user-images.githubusercontent.com/50188795/159826248-bffeccf1-43ac-4c3a-ad81-7c86fb1f760d.png)

다음과 같이 테스트 해봤으나 정상 동작 하지 않았습니다 

어떻게 동작하게 했나요?

**Reply by pumfit** [reply]
저도 작성자분이랑 거의 동일한 프로세스로 진행하였습니다. 아래는 코드 전문이고
```
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class EventFunctions extends ZepetoScriptBehaviour {

    OnMouseDown() {
        console.log('OnMouseDown.');
    }
    OnMouseDrag() {
        console.log('OnMouseDrag.');
    }  
    OnMouseUp() {
        console.log('OnMouseUp.');
    }  
    OnMouseEnter() {
        console.log('OnMouseEnter.');
    } 
    OnMouseExit() {
        console.log('OnMouseExit.');
    }  
    OnMouseOver() {
        console.log('OnMouseOver.');
    }  
    OnMouseUpAsButton() {
        console.log('OnMouseUpAsButton.');
    } 
}
```
현재 진행한 버전이고
![image](https://user-images.githubusercontent.com/46295539/159827417-e1843751-9256-4431-8a61-8e8e212bffc3.png)

아래는 동작하는 영상입니다!
https://user-images.githubusercontent.com/46295539/159827270-59c59338-fc1b-4653-bbbe-89c97f73f578.mov

**Reply by smilejsu82** [reply]
유니티 2020.3.9f1 버전을 선택해 제페토 템플릿으로 새로운 프로젝트를 생성 했습니다 
![image](https://user-images.githubusercontent.com/50188795/159828555-4a43f123-a6cc-4584-97e1-6a51dde3aa01.png)

그리고 새로운 인풋 시스템을 위한 팝업에 YES를 선택 했습니다 
![image](https://user-images.githubusercontent.com/50188795/159828643-10fb12c9-1592-4465-a9d3-29e31545bb19.png)

패키지 매니저에서 확인해본결과 ZEPETO.World가 이미 1.1.5버전 인것을 확인했습니다 
![image](https://user-images.githubusercontent.com/50188795/159828798-adfb2557-b91c-4bc7-917e-069809f8134f.png)

최초 제페토 환경 설정을 했을때 ZEPETO.World버전은 1.1.2였습니다 
![image](https://user-images.githubusercontent.com/50188795/159828840-fa31e8c0-778b-42e5-8b9f-413602891e6f.png)
![image](https://user-images.githubusercontent.com/50188795/159828935-6aa36e73-8c8a-4803-8c83-d2f64435d405.png)

하지만 한번 업데이트 하고 나면 1.1.5를 유지 하는군요 

이것이 문제인가요?

새 오브젝트 (CUBE)를 만들고 
![image](https://user-images.githubusercontent.com/50188795/159829029-2a73deda-2fc7-4099-874b-2ad8b8b9b2f3.png)

스크립트를 생성 하고 이벤트 함수를 작성 했습니다 
![image](https://user-images.githubusercontent.com/50188795/159829283-0fdeeb49-ef15-4b18-8305-7a57e6d79805.png)
![image](https://user-images.githubusercontent.com/50188795/159829307-048c2195-889a-4589-86ec-3a5dffd3f974.png)

그리고 오브젝트에 부착 한후 테스트 해봤습니다 

https://user-images.githubusercontent.com/50188795/159829970-f2deb2aa-96db-4a69-8938-6d4388ad5756.mp4

하지만 동작하지 않네요 

다음은 제페토 프로젝트가 아닌 일반 프로젝트 (3D 템플릿)에서 테스트한 결과 입니다 
이것은 정상 동작 합니다 

https://user-images.githubusercontent.com/50188795/159836683-b466a891-f394-4cc7-88c6-b7286fad1f22.mp4


무엇을 잘못한것일까요?

**Comment by smilejsu82** [comment]
제페토 템플릿으로 프로젝트를 열때 새로운 인풋시스템에 대한 팝업이 나오는데 이것은 정확하게 무엇을 의미하나요?

![image](https://user-images.githubusercontent.com/50188795/159838574-3fe22e15-c896-4fb0-a701-de22b874b41f.png)

![image](https://user-images.githubusercontent.com/50188795/159838613-e577772a-d28c-46c2-ac35-3c8d7ea0f422.png)

NO를 선택하고나서 OnMouseXXX 이벤트 함수가 정상적으로 동작 했습니다 

![image](https://user-images.githubusercontent.com/50188795/159838091-4b919aaa-e3e9-469a-b792-33324dc850da.png)

**Reply by Zepeto-TechEvangelist** [reply]
사용하시는 VSCode에디터에 밑줄이 생기는 것으로 보아 에디터 문제일 수도 있겠습니다.
아래 가이드대로 적용 부탁 드립니다.

https://studio.zepeto.me/kr/guides/preparing-for-world-creating
